firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionTesterApiClient.java

```diff
@@ -0,0 +1,131 @@
+package com.google.firebase.appdistribution;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import javax.net.ssl.HttpsURLConnection;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class FirebaseAppDistributionTesterApiClient {
+
+  private final String fid;
+  private final String appId;
+  private final String apiKey;
+
+  private final String TAG = "FADTesterApiClient";
+
+  private static final String RELEASE_ENDPOINT_URL_FORMAT =
+      "https://firebaseapptesters.googleapis.com/v1alpha/devices/-/testerApps/%s/installations/%s/releases";
+  private static final String REQUEST_METHOD = "GET";
+  private static final String API_KEY_HEADER = "x-goog-api-key";
+  private static final String INSTALLATION_AUTH_HEADER = "X-Goog-Firebase-Installations-Auth";
+  private static final String BUILD_VERSION_JSON_KEY = "buildVersion";
+  private static final String DISPLAY_VERSION_JSON_KEY = "displayVersion";
+  private static final String RELEASE_NOTES_JSON_KEY = "releaseNotes";
+  private static final String BINARY_TYPE_JSON_KEY = "binaryType";
+  public static final int DEFAULT_BUFFER_SIZE = 8192;
```

Comment:
How did we choose this number?

Reply:
It was used in the examples I found for converting the input stream to string, and also is the default used by BufferedReader (http://www.docjar.com/html/api/java/io/BufferedReader.java.html) which is what I wanted to use originally but was incompatible with certain API levels.

Reply:
Awesome :)